tests/components/blue_current/test_init.py

```diff
@@ -0,0 +1,185 @@
+                "actual_p3": 15,
+                "activity": "charging",
+                "start_datetime": "2021-11-18T14:12:23",
+                "stop_datetime": "2021-11-18T14:32:23",
+                "offline_since": "2021-11-18T14:32:23",
+                "total_cost": 10.52,
+                "vehicle_status": "standby",
+                "actual_kwh": 10,
+            }
+        }
+
+        test_async_dispatcher_send.assert_called_with(
+            hass, "blue_current_value_update_101"
+        )
+
+        # test GRID_STATUS
+        data3 = {
+            "object": "GRID_STATUS",
+            "data": {
+                "grid_actual_p1": 12,
+                "grid_actual_p2": 14,
+                "grid_actual_p3": 15,
+            },
+        }
+        await connector.on_data(data3)
+        assert connector.grid == {
+            "grid_actual_p1": 12,
+            "grid_actual_p2": 14,
+            "grid_actual_p3": 15,
+        }
+        test_async_dispatcher_send.assert_called_with(hass, "blue_current_grid_update")
+
+
+async def test_start_loop(hass: HomeAssistant) -> None:
+    """Tests start_loop."""
+
+    with patch(
+        "homeassistant.components.blue_current.async_call_later"
+    ) as test_async_call_later:
+        config_entry = MockConfigEntry(
+            domain=DOMAIN,
+            entry_id="uuid",
+            unique_id="uuid",
+            data={"api_token": "123", "card": {"123"}},
+        )
+
+        connector = Connector(hass, config_entry, Client)
+
+        with patch(
+            "bluecurrent_api.Client.start_loop",
+            side_effect=WebsocketError("unknown command"),
+        ):
+            await connector.start_loop()
+            test_async_call_later.assert_called_with(hass, 1, connector.reconnect)
+
+        with patch(
+            "bluecurrent_api.Client.start_loop", side_effect=RequestLimitReached
+        ):
+            await connector.start_loop()
+            test_async_call_later.assert_called_with(hass, 1, connector.reconnect)
+
+
+async def test_reconnect(hass: HomeAssistant) -> None:
+    """Tests reconnect."""
+
+    with patch("bluecurrent_api.Client.connect"), patch(
+        "bluecurrent_api.Client.connect", side_effect=WebsocketError
+    ), patch(
+        "bluecurrent_api.Client.get_next_reset_delta", return_value=timedelta(hours=1)
+    ), patch(
+        "homeassistant.components.blue_current.async_call_later"
+    ) as test_async_call_later:
+        config_entry = MockConfigEntry(
+            domain=DOMAIN,
+            entry_id="uuid",
+            unique_id="uuid",
+            data={"api_token": "123", "card": {"123"}},
+        )
+
+        connector = Connector(hass, config_entry, Client)
+        await connector.reconnect()
+
+        test_async_call_later.assert_called_with(hass, 20, connector.reconnect)
+
+        with patch("bluecurrent_api.Client.connect", side_effect=RequestLimitReached):
+            await connector.reconnect()
```

Comment:
Move time forward instead to test the reconnection.

We have this helper:
https://github.com/home-assistant/core/blob/8140036d2e72290ba0bcc9b1c9517520622255a1/tests/common.py#L412

Possibly combine it with freezing time.